### PR TITLE
[enhancement] get valuation rate and gross profit on sales order item

### PR DESCRIPTION
**Pull request by saurabh6790**

### CODE REVIEW

File: erpnext/stock/get_item_details.py [hunk 1]
```diff
@@ -68,6 +67,8 @@ def get_item_details(args):
 
 	if args.get("is_subcontracted") == "Yes":
 		out.bom = get_default_bom(args.item_code)
+		
+	get_goss_profit(out)
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: spelling?

File: erpnext/stock/get_item_details.py [hunk 2]
```diff
@@ -463,3 +467,10 @@ def get_default_bom(item_code=None):
 			return bom
 		else:
 			frappe.throw(_("No default BOM exists for Item {0}").format(item_code))
+			
+			
+def get_goss_profit(out):
+	out.update({
+		"gross_profit": ((out.price_list_rate - out.valuation_rate) * out.qty) * (out.conversio_rate or 1)
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: spelling of conversion_rate?

gross_profit = (out.base_rate - out.valuation_rate) * out.qty

It is in company_currency

File: erpnext/selling/doctype/sales_order_item/sales_order_item.json [hunk 3]
```diff
@@ -1222,6 +1222,56 @@
...
+   "search_index": 0, 
+   "set_only_once": 0, 
+   "unique": 0
+  }, 
+  {
+   "allow_on_submit": 0, 
+   "bold": 0, 
+   "collapsible": 0, 
+   "fieldname": "gross_profit", 
+   "fieldtype": "Currency",
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order_item/sales_order_item.json]: set options to show company currency symbol in gross profit value

File: erpnext/public/js/controllers/transaction.js [hunk 4]
```diff
@@ -866,6 +867,10 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 				refresh_field('to_date');
 			}
 		}
+	},
+	
+	gross_profit: function(item) {
```
Comment (nabinhait) [inline erpnext/public/js/controllers/transaction.js]: set_gross_profit()

correct the formulae

and it is only applicable to sales order

Comment (saurabh6790) [inline erpnext/public/js/controllers/transaction.js]: @nabinhait 

Not getting value of base_rate fields from js side on rate trigger if transaction is in company currency.

**Comment by saurabh6790** [comment]
@nabinhait 

fixed calculation for non-stock and product bundle items

**Comment by anandpdoshi** [comment]
needs to be fixed as per discussion with @nabinhait

**Comment by saurabh6790** [comment]
@anandpdoshi  Fixed. as per discussions.

**Pull request merged**